llvmlite/ir/values.py

```diff
@@ -14,13 +14,15 @@
 
 _VALID_CHARS = (frozenset(map(ord, string.ascii_letters)) |
                 frozenset(map(ord, string.digits)) |
-                frozenset('._-$'))
+                frozenset(map(ord, ' !#$%&\'()*+,-./:;<=>?@[]^_`{|}~')))
```

Comment:
Hmm, I understand. Perhaps `_map` can contain both cases on py2 instead? That way, the branch would happen at construction time. Something like:
```
if six.PY2:
    _map[chr(ch)] = _map[ch]
```